lib/nodes/orchestra.cpp

```diff
@@ -0,0 +1,679 @@
+      ddf.writeToFile(*dataDefinitionFilename);
+
+      logger->info("Wrote Orchestra Data Definition file (DDF) to '{}'",
+                   *dataDefinitionFilename);
+
+      if (dataDefinitionFileWriteOnly) {
+        return Node::prepare();
+      }
+    }
+
+    logger->debug("Connecting to Orchestra framework: domain={}, ddf={}, "
+                  "connection_key={}",
+                  domain.name,
+                  dataDefinitionFilename ? *dataDefinitionFilename : "none",
+                  connectionKey);
+
+    RTConnectionLockGuard guard(connectionKey);
+
+    auto ret = dataDefinitionFilename
+                   ? RTConnectWithFile(dataDefinitionFilename->c_str(),
+                                       domain.name.c_str(), connectTimeout)
+                   : RTConnect(domain.name.c_str(), connectTimeout);
+    if (ret != RTAPI_SUCCESS) {
+      throw RTError(ret, "Failed to connect to Orchestra framework");
+    }
+
+    logger->info("Connected to Orchestra framework: domain={}", domain.name);
+
+    ret = RTGetConnectionStatusPtr(connectionKey, &status);
+    if (ret != RTAPI_SUCCESS) {
+      throw RTError(ret, "Failed to get connection status pointer");
+    }
+
+    ret = RTDefineFlagDelay(flagDelay);
+    if (ret != RTAPI_SUCCESS) {
+      throw RTError(ret, "Failed to set flag delay");
+    }
+
+    ret = RTUseFlagWithTool(useFlagWithTool);
+    if (ret != RTAPI_SUCCESS) {
+      throw RTError(ret, "Failed to set flag with tool");
+    }
+
+    ret = RTSetSkipWaitToGoAtConnection(skipWaitToGo);
+    if (ret != RTAPI_SUCCESS) {
+      throw RTError(ret, "Failed to check ready to go");
+    }
+
+    if (auto *c = dynamic_cast<ConnectionRemote *>(domain.connection)) {
+      ret = RTSetupRefMemRemoteConnection(c->name.c_str(), c->pciIndex);
+      if (ret != RTAPI_SUCCESS) {
+        throw RTError(ret,
+                      "Failed to setup reflective memory remote connection");
+      }
+    }
+
+    for (auto &[item, mapping] : publishMappings) {
+      mapping.prepare(connectionKey);
+    }
+
+    for (auto &[item, mapping] : subscribeMappings) {
+      mapping.prepare(connectionKey);
+    }
+
+    char *bufferPublish, *bufferSubscribe;
+    unsigned int bufferPublishSize, bufferSubscribeSize;
+    ret = RTGetBufferInfo(connectionKey, &bufferPublish, &bufferPublishSize,
+                          &bufferSubscribe, &bufferSubscribeSize);
+    if (ret != RTAPI_SUCCESS) {
+      throw RTError(ret, "Failed to get buffer info");
+    }
+    logger->debug("Buffer publish:   size={}, ptr={:p}", bufferPublishSize,
+                  bufferPublish);
+    logger->debug("Buffer subscribe: size={}, ptr={:p}", bufferSubscribeSize,
+                  bufferSubscribe);
+
+    return Node::prepare();
+  }
+
+  int start() override {
+    if (dataDefinitionFileWriteOnly) {
+      return Node::start();
+    }
+
+    RTConnectionLockGuard guard(connectionKey);
+
+    RTWaitReadyToGo();
+
+    if (!domain.synchronous) {
```

Comment:
maybe rename to asynchronous, because I only saw negative checks, and positive checks are more readable.

Reply:
This is an external setting. I would like to keep the semantics aligned with the Orchestra software here.